python/lib/dicom/summary_make.py

```diff
@@ -0,0 +1,218 @@
+from functools import cmp_to_key
+import os
+import pydicom
+import pydicom.errors
+from lib.dicom.summary_type import Summary, Info, Patient, Scanner, Acquisition, DicomFile, OtherFile
+from lib.dicom.text import make_hash, read_dicom_date_none
+
+
+def get_value(dicom: pydicom.Dataset, tag: str):
+    """
+    Get a required value from a DICOM.
+    """
+    if tag not in dicom:
+        raise Exception(f'Expected DICOM tag \'{tag}\' but found none.')
+
+    return dicom[tag].value
+
+
+def get_value_none(dicom: pydicom.Dataset, tag: str):
+    """
+    Get a nullable value from a DICOM.
+    """
+    if tag not in dicom:
+        return None
+
+    return dicom[tag].value or None
+
+
+def cmp_int_none(a: int | None, b: int | None):
+    """
+    Order comparison between two nullable integers.
+    """
+    match a, b:
```

Comment:
Both options are fine with me, tell me if you have a preference. If we want to be really rigorous though, it might be good to update as 3.8 should (theoretically) soon become EOL.

https://devguide.python.org/versions/